gatsby-node.js

```diff
@@ -84,6 +92,79 @@ const constructDbFilter = () => ({
     commit_hash: process.env.COMMIT_HASH || { $exists: false },
 });
 
+const createTagPageType = async (createPage, pageMetadata, stitchType) => {
+    const isAuthor = stitchType === 'author';
+    const pageType = STITCH_TYPE_TO_URL_PREFIX[stitchType];
+
+    // Query for all possible values for this type of tag
+    const possibleTagValues = await stitchClient.callFunction(
+        'getValuesByKey',
+        [metadata, stitchType]
+    );
+
+    const requests = [];
+
+    // For each possible tag value, query the pages that exist for it
+    await possibleTagValues.forEach(async tag => {
+        const requestKey = {};
+        requestKey[stitchType] = tag._id;
+        requests.push(
+            stitchClient.callFunction('fetchDevhubMetadata', [
+                metadata,
+                requestKey,
+            ])
+        );
+    });
+
+    const pageData = await Promise.all(requests);
+
+    // Once requests finish, map the item with name (and optional image) to the response's return value
+    const itemsWithPageData = possibleTagValues.map((r, i) => ({
+        item: r,
+        pages: pageData[i],
+    }));
+
+    const PAGES = [];
```

Comment:
```suggestion
    const pageList = [];
```

Reply:
I might even suggest a map right on this variable below rather than a forEach, but it's a nit